crates/core/src/table/builder.rs

```diff
@@ -202,13 +202,26 @@ impl OptionResolver {
             if Storage::CLOUD_STORAGE_PREFIXES
                 .iter()
                 .any(|prefix| key.starts_with(prefix))
-                && !self.storage_options.contains_key(&key.to_ascii_lowercase())
             {
-                self.storage_options.insert(key.to_ascii_lowercase(), value);
+                if let Some(storage_key) = self.process_cloud_env_var(&key) {
+                    if !self.storage_options.contains_key(&storage_key) {
+                        self.storage_options.insert(storage_key, value);
+                    }
+                }
             }
         }
     }
 
+    /// Process a cloud environment variable and return the appropriate storage key
+    fn process_cloud_env_var(&self, env_key: &str) -> Option<String> {
+        if env_key.starts_with("HOODIE_ENV_fs_DOT_s3a_DOT_") || env_key.starts_with("HOODIE_ENV_fs_DOT_s3_DOT_") {
```

Comment:
this does not seem right. why are these 2 strings checked in particular? shouldn't it be just `HOODIE_ENV_` ?